compiler/rustc_expand/src/expand.rs

```diff
@@ -1042,6 +1042,9 @@ trait InvocationCollectorNode: HasAttrs + HasNodeId + Sized {
     fn expand_cfg_false(&mut self, collector: &mut InvocationCollector<'_, '_>, span: Span) {
         collector.cx.emit_err(RemoveNodeNotSupported { span, descr: Self::descr() });
     }
+    fn declared_names(&self) -> Vec<Ident> {
```

Comment:
This needs a comment saying that it's for diagnostics only, it's an approximation that cannot be used for anything precise.